### PR TITLE
fix: support custom content-type (#96)

### PR DESCRIPTION
**Pull request by mebtte**
#96

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -136,7 +136,10 @@ async function send (ctx, path, opts = {}) {
     }
     ctx.set('Cache-Control', directives.join(','))
   }
-  ctx.type = type(path, encodingExt)
+
+  // support custom content-type
+  if (!ctx.type)  ctx.type = type(path, encodingExt)
+
```
Comment (coderhaoxin) [inline index.js]: Please add a test case for this :)

Comment (mebtte) [reply index.js]: ok

Comment (mebtte) [reply index.js]: it's done.

**Comment by coderhaoxin** [comment]
@mebtte 

If we set body first, such as

```js
ctx.body = {
  message: 'hello'
}
```

koa will set `type = 'json'` automatically,
then we call `send(ctx, filename)`, we should correct the `content-type` :)

**Comment by coderhaoxin** [comment]
For your use cases, just put `ctx.type = 'bin'` after `send` :)

```js
await send(ctx, '...')
ctx.type = 'bin'
```

Sorry for the mistake 😢

**Comment by mebtte** [comment]
oo

**Pull request closed without merging**